Description: When building --with-cpu=power7, tune for power8
Author: Adam Conrad <adconrad@ubuntu.com>

--- eglibc-2.19.orig/configure
+++ eglibc-2.19/configure
@@ -6873,6 +6873,10 @@ if ${libc_cv_cc_submachine+:} false; the
 else
     libc_cv_cc_submachine=no
   for opt in "-march=$submachine" "-mcpu=$submachine"; do
+    # dirty hack to tune all power7 builds for power8:
+    if [ "$opt" = "-mcpu=power7" ]; then
+      opt="-mcpu=power7 -mtune=power8"
+    fi
     if { ac_try='${CC-cc} $opt -xc /dev/null -S -o /dev/null'
   { { eval echo "\"\$as_me\":${as_lineno-$LINENO}: \"$ac_try\""; } >&5
   (eval $ac_try) 2>&5
--- eglibc-2.19.orig/configure.ac
+++ eglibc-2.19/configure.ac
@@ -1921,6 +1921,10 @@ if test -n "$submachine"; then
 		 libc_cv_cc_submachine, [dnl
   libc_cv_cc_submachine=no
   for opt in "-march=$submachine" "-mcpu=$submachine"; do
+    # dirty hack to tune all power7 builds for power8:
+    if [ "$opt" = "-mcpu=power7" ]; then
+      opt="-mcpu=power7 -mtune=power8"
+    fi
     LIBC_TRY_CC_OPTION([$opt], [
       libc_cv_cc_submachine="$opt"
       break], [])
