Index: eglibc-2.18/sysdeps/hppa/fpu/feholdexcpt.c
===================================================================
--- eglibc-2.18.orig/sysdeps/hppa/fpu/feholdexcpt.c	2013-06-15 13:37:04.000000000 -0400
+++ eglibc-2.18/sysdeps/hppa/fpu/feholdexcpt.c	2014-04-09 21:01:32.492451011 -0400
@@ -29,8 +29,8 @@
   /* Store the environment.  */
   bufptr = clear.buf;
   __asm__ (
-	   "fstd,ma %%fr0,8(%1)\n"
-	   : "=m" (clear), "+r" (bufptr) : : "%r0");
+	   "fstd %%fr0,0(%1)\n"
+	   : "=m" (clear) : "r" (bufptr) : "%r0");
   memcpy (envp, &clear.env, sizeof (fenv_t));
 
   /* Clear exception queues */
@@ -44,7 +44,7 @@
      Thus we start bufptr at the end and work backwards */
   bufptr = (unsigned long long *)((unsigned int)(clear.buf) + sizeof(unsigned int)*4);
   __asm__ (
-	   "fldd,mb -8(%0),%%fr0\n"
+	   "fldd 0(%0),%%fr0\n"
 	   : : "r" (bufptr), "m" (clear) : "%r0");
 
   return 0;
Index: eglibc-2.18/sysdeps/hppa/fpu/fesetenv.c
===================================================================
--- eglibc-2.18.orig/sysdeps/hppa/fpu/fesetenv.c	2013-06-15 13:37:04.000000000 -0400
+++ eglibc-2.18/sysdeps/hppa/fpu/fesetenv.c	2014-04-09 20:59:45.221386729 -0400
@@ -33,7 +33,7 @@
      we want to use from the environment specified by the parameter.  */
   bufptr = temp.buf;
   __asm__ (
-	   "fstd,ma %%fr0,8(%1)\n"
+	   "fstd %%fr0,0(%1)\n"
 	   : "=m" (temp) : "r" (bufptr) : "%r0");
 
   temp.env.__status_word &= ~(FE_ALL_EXCEPT
@@ -54,7 +54,7 @@
      we take advantage of that to load in reverse order so fr0
      is loaded last and T-Bit is enabled. */
   __asm__ (
-	   "fldd,mb -8(%1),%%fr0\n"
+	   "fldd 0(%1),%%fr0\n"
 	   : : "m" (temp), "r" (bufptr) : "%r0" );
 
   /* Success.  */
