commit 9300797ee93422a87f443d5ba7e1411167e7a837
Author: Ondřej Bílka <neleai@seznam.cz>
Date:   Tue Jan 6 17:43:09 2015 +0100

    Suppress warning in string/tester.c for gcc 4.9

diff --git a/string/tester.c b/string/tester.c
index f957ed2..64dc056 100644
--- a/string/tester.c
+++ b/string/tester.c
@@ -1305,10 +1305,10 @@ test_memset (void)
   equal(one, "axxxefgh", 2);		/* Basic test. */
 
   DIAG_PUSH_NEEDS_COMMENT;
-#if __GNUC_PREREQ (5, 0)
-  /* GCC 5.0 warns about a zero-length memset because the arguments to memset
+#if __GNUC_PREREQ (4, 9)
+  /* GCC 4.9 warns about a zero-length memset because the arguments to memset
      may be in the wrong order.  But we really want to test this.  */
-  DIAG_IGNORE_NEEDS_COMMENT (5.0, "-Wmemset-transposed-args")
+  DIAG_IGNORE_NEEDS_COMMENT (4.9, "-Wmemset-transposed-args")
 #endif
   (void) memset(one+2, 'y', 0);
   equal(one, "axxxefgh", 3);		/* Zero-length set. */
